features/base/tracks/middleware.js

```diff
@@ -2,29 +2,71 @@ import {
     LIB_DISPOSED,
     LIB_INITIALIZED
 } from '../lib-jitsi-meet';
+import {
+    CAMERA_FACING_MODE_CHANGED,
+    VIDEO_MUTED_STATE_CHANGED,
+    MEDIA_TYPE,
+    AUDIO_MUTED_STATE_CHANGED
+} from '../media';
 import { MiddlewareRegistry } from '../redux';
 
 import {
     createLocalTracks,
     destroyLocalTracks
 } from './actions';
+import {
+    getLocalTrack,
+    setTrackMuted
+} from './functions';
 
 /**
  * Middleware that captures LIB_INITIALIZED and LIB_DISPOSED actions
- * and respectively creates/destroys local media tracks.
+ * and respectively creates/destroys local media tracks. Also listens to media-
+ * related actions and performs corresponding operations with tracks.
  *
  * @param {Store} store - Redux store.
  * @returns {Function}
  */
 MiddlewareRegistry.register(store => next => action => {
     switch (action.type) {
+    case AUDIO_MUTED_STATE_CHANGED: {
+        const tracks = store.getState()['features/base/tracks'];
+
+        setTrackMuted(
+            (getLocalTrack(tracks, MEDIA_TYPE.AUDIO) || {}).jitsiTrack,
```

Comment:
I don't like `getLocalTrack(tracks, MEDIA_TYPE.AUDIO) || {}` because (1) it creates an empty object to get a property which is known to not exist on that object and (2) setTrackMuted doesn't need to be invoked in that case anyway.